integration-tests-jvm/flink/src/main/java/org/apache/camel/quarkus/component/flink/it/FlinkResource.java

```diff
@@ -33,18 +48,69 @@ public class FlinkResource {
     private static final Logger LOG = Logger.getLogger(FlinkResource.class);
 
     private static final String COMPONENT_FLINK = "flink";
+
     @Inject
     CamelContext context;
 
-    @Path("/load/component/flink")
-    @GET
+    @Inject
+    ProducerTemplate template;
+
+    String flinkDataSetUri = "flink:dataSet?dataSet=#myDataSet";
+    String flinkDataStreamUri = "flink:datastream?datastream=#myDataStream";
+
+    @Path("/dataset/{fileName}")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
     @Produces(MediaType.TEXT_PLAIN)
-    public Response loadComponentFlink() throws Exception {
-        /* This is an autogenerated test */
-        if (context.getComponent(COMPONENT_FLINK) != null) {
-            return Response.ok().build();
+    public Response dataSetFromTextFile(@PathParam("fileName") String fileName) {
+
+        if (Files.exists(Paths.get(fileName))) {
+            ExecutionEnvironment env = Flinks.createExecutionEnvironment();
+            DataSet<String> myDataSet = env.readTextFile(fileName);
+            context.getRegistry().bind("myDataSet", myDataSet);
+            context.getRegistry().bind("countTotal", addDataSetCallback());
+            Long totalCount = template.requestBody(
+                    flinkDataSetUri + "&dataSetCallback=#countTotal", null, Long.class);
+            return Response.ok(totalCount).build();
         }
-        LOG.warnf("Could not load [%s] from the Camel context", COMPONENT_FLINK);
-        return Response.status(500, COMPONENT_FLINK + " could not be loaded from the Camel context").build();
+
+        return Response.status(Response.Status.NOT_FOUND).build();
     }
+
+    @Path("/datastream")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response loadStream(String data) throws IOException {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        DataStream<String> datastream = env.fromElements(data);
+        context.getRegistry().bind("myDataStream", datastream);
+        java.nio.file.Path output;
+        output = Files.createTempFile("camel", "txt");
```

Comment:
Instead of creating the file here, it would be better to do like the `dataSetCallback` test. Create the file (and later delete it) in the test method & pass the path to the JAX-RS endpoint.